### PR TITLE
CA-267058: hidden password for iscsi and cifs.

### PR DESCRIPTION
**Pull request by lln133208**
2187297 for NFSv4
2187298 for NFS
2187299 for HBA
2187300 waiting hosts to run
Signed-off-by: Yuan Ren <yuan.ren@citrix.com>

### CODE REVIEW

File: src/XenCert/StorageHandler.py [hunk 1]
```diff
@@ -2138,24 +2136,25 @@ def Create(self, device_config = {}):
                 device_config['targetIQN'] = '*'
             else:
                 device_config['targetIQN'] = self.iqn
-            if self.storage_conf['chapuser']!= None and self.storage_conf['chappasswd'] != None:
+            if self.storage_conf['chapuser']is not None and self.storage_conf['chappasswd'] is not None:
```
Comment (DeliZhangX) [inline src/XenCert/StorageHandler.py]: need a sapce

File: src/XenCert/StorageHandler.py [hunk 2]
```diff
@@ -2138,24 +2136,25 @@ def Create(self, device_config = {}):
...
             # try to create an SR with one of the LUNs mapped, if all fails throw an exception
             for scsiId in listSCSIId:
                 try:                    
                     device_config['SCSIid'] = scsiId
-                    XenCertPrint("The SR create parameters are %s, %s" % (util.get_localhost_uuid(self.session), device_config))
+                    device_config_tmp = XenCertCommon.getNoPasswordConfig(device_config, self.storage_conf['storage_type'])
+                    XenCertPrint("The SR create parameters are %s, %s" % (util.get_localhost_uuid(self.session), device_config_tmp))
                     sr_ref = self.session.xenapi.SR.create(util.get_localhost_uuid(self.session), device_config, '0', 'XenCertTestSR', '', 'lvmoiscsi', '',True, {})
-                    XenCertPrint("Created the SR %s using device_config %s" % (sr_ref, device_config))
+                    XenCertPrint("Created the SR %s using device_config %s" % (sr_ref, device_config_tmp))
```
Comment (DeliZhangX) [inline src/XenCert/StorageHandler.py]: Just a suggestion: needless to print device_config_tmp again here.
    XenCertPrint("Created the SR %s" % sr_ref )

File: src/XenCert/StorageHandler.py [hunk 3]
```diff
@@ -3077,16 +3077,17 @@ def Create(self):
         try:
             # Create an SR on the CIFS server/share provided.
             Print("      Creating the SR.")
-            XenCertPrint("The SR create parameters are %s, %s" % (util.get_localhost_uuid(self.session), device_config))
+            device_config_tmp = XenCertCommon.getNoPasswordConfig(device_config, self.storage_conf['storage_type'])
+            XenCertPrint("The SR create parameters are %s, %s" % (util.get_localhost_uuid(self.session), device_config_tmp))
             sr_ref = self.session.xenapi.SR.create(util.get_localhost_uuid(self.session), device_config, '0', 'XenCertTestSR', '', 'cifs', '',False, {})
-            XenCertPrint("Created the SR %s using device_config %s" % (sr_ref, device_config))
+            XenCertPrint("Created the SR %s using device_config %s" % (sr_ref, device_config_tmp))
```
Comment (DeliZhangX) [inline src/XenCert/StorageHandler.py]: same to above

File: src/XenCert/XenCertCommon.py [hunk 4]
```diff
@@ -274,15 +274,25 @@ def printCommand(argvs):
...
+                temp_argvs[option_index+1] = ':'.join(getNoPasswordCmds(temp_argvs[option_index+1].split(':'), 2))
             else:
                 temp_argvs[option_index+1] = HIDDEN_PASSWORD
 
     for argv in temp_argvs:
         PrintToLog(argv)
         PrintToLog(' ')
 
-def hidePassword(cmd, password_index=-3):
+def getNoPasswordCmds(cmd, password_index=-3):
```
Comment (DeliZhangX) [inline src/XenCert/XenCertCommon.py]: Suggest to change to another name, e.g. getCmdsWithHiddenPassword

File: src/XenCert/XenCertCommon.py [hunk 5]
```diff
@@ -274,15 +274,25 @@ def printCommand(argvs):
...
     cmd_with_hidden_password = cmd[:]
     cmd_with_hidden_password[password_index] = HIDDEN_PASSWORD
     return cmd_with_hidden_password
+
+def getNoPasswordConfig(config, storage_type):
```
Comment (DeliZhangX) [inline src/XenCert/XenCertCommon.py]: same to above

File: src/XenCert/XenCertCommon.py [hunk 6]
```diff
@@ -274,15 +274,25 @@ def printCommand(argvs):
...
+    config_with_hidden_password = dict(config)
+    if storage_type == 'iscsi' and config.get('chappassword'):
```
Comment (DeliZhangX) [inline src/XenCert/XenCertCommon.py]: It's good to check storage_type indeed here. while I still suggest to remove storage_type, just need
if config.get('chappassword') is not None:
    config_with_hidden_password['chappassword'] = HIDDEN_PASSWORD
if config.get('password') is not None:
    config_with_hidden_password['password'] = HIDDEN_PASSWORD
It' will make simpler, also hidden empty "" password.

File: src/XenCert/XenCertCommon.py (hunk 4 above)
Comment (lln133208) [reply src/XenCert/XenCertCommon.py]: native english expression :+1:

**Comment by lln133208** [comment]
@DeliZhangX Rework has been finished.

**Comment by lln133208** [comment]
Thank you

**Pull request merged**